nmostesting/mocks/Node.py

```diff
@@ -78,3 +193,261 @@ def node_video_sdp(stream_type):
+        'destination_port': sender['activations']['transport_params'][0]['destination_port'] if master_enable and sender else default_destination_port,
+        'source_ip': sender['activations']['transport_params'][0]['source_ip'] if master_enable and sender else None,
+        'interface_ip': get_default_ip() if master_enable and sender else default_interface_ip,
+        'rtp_enabled': True
+    }
+
+    transport_params = receiver.get('transport_params') if master_enable and receiver else None
+    transport_file = receiver.get('transport_file') if master_enable and receiver and 'transport_file' in receiver else {'data': None, 'type': None}
+    sender_id = receiver.get('sender_id') if master_enable and receiver else None
+
+    updated_transport_params = dict(transport_params[0], **transport_params_update) if transport_params else transport_params_update
+
+    activation_update = {
+        "activation": {
+            "activation_time": str(time.time()).replace('.', ':') if master_enable and activation else None, 
+            "mode": activation['mode'] if master_enable and activation else None, 
+            "requested_time": None
+        },
+        'master_enable': master_enable,
+        'sender_id': sender_id,
+        'transport_file': transport_file,
+        'transport_params': [ updated_transport_params ]
+    }
+
+    return activation_update
+
+def _update_receiver_subscription(receiver, active, sender_id):
+
+    receiver_update = {
+        'subscription': {'active': active, 'sender_id': sender_id},
+        'version': NMOSUtils.get_TAI_time()
+    }
+
+    return dict(receiver, **receiver_update)
+
+def _update_sender_subscription(sender, active):
+
+    sender_update = {
+        'subscription': {'active': active, 'receiver_id': None},
+        'version': NMOSUtils.get_TAI_time()
+    }
+
+    return dict(sender, **sender_update)
+
+@NODE_API.route('/x-nmos/connection/<version>/single/<resource>/<resource_id>/staged', methods=["GET", "PATCH"], strict_slashes=False)
+def staged(version, resource, resource_id):
+    """
+    GET returns current staged data for given resource
+    PATCH updates data for given resource, either staging a connection, activating a staged connection, 
+    activating a connection without staging or deactivating an active connection
+    Updates data then POSTs updated receiver to registry
+    """
+    NODE.staged_requests.append({'method': request.method, 'resource': resource, 'resource_id': resource_id, 'data': request.json})
+
+    try:
+        if resource == 'senders':
+            resources = NODE.senders
+        
+            if request.method == 'PATCH':
+
+                activations = resources[resource_id]['activations']
+                sender = resources[resource_id]['sender']
+
+                activations['active']['master_enable'] = request.json.get("master_enable", True)
+                activations['active']['activation']['activation_time'] = str(time.time()).replace('.', ':')
```

Comment:
This time formatting comes up in a few places. I doubt it'll cause specific issues for this testing, but it should really be formatted in nanoseconds. Py3.7 has a time_ns function, but zero padding as necessary would be fine. I'm not expecting handling of leap seconds...

Reply:
I thought we had a utility function to make TAI `<seconds>:<nanoseconds>` timestamps, e.g. to do absolute activations, but maybe I misremembered.

Reply:
@jonathan-r-thorpe this one:
https://github.com/AMWA-TV/nmos-testing/blob/0513479c15c412a66be51ae6604a78120112e9bc/nmostesting/NMOSUtils.py#L90

Reply:
OK noted